modules/private-shared-lib/src/events/outbound_command_event/process_party_info_callback.ts

```diff
@@ -29,38 +29,45 @@ import { IMessageHeader } from '@mojaloop/platform-shared-lib-messaging-types-li
 import { IPartyResult } from '../../types';
 
 export interface IProcessPartyInfoCallbackCmdEvtData {
-    key: string;
-    partyResult: IPartyResult;
+    bulkId: string;
+    content: {
+        transferId: string;
+        partyResult: IPartyResult;
+    };
     timestamp: number | null;
     headers: IMessageHeader[] | null;
 }
 
 export class ProcessPartyInfoCallbackCmdEvt extends CommandEvent {
     constructor(data: IProcessPartyInfoCallbackCmdEvtData) {
         super({
-            key: data.key,
-            content: data.partyResult,
+            key: data.bulkId,
+            content: data.content,
             timestamp: data.timestamp,
             headers: data.headers,
             name: ProcessPartyInfoCallbackCmdEvt.name,
         });
     }
 
     getBulkId() {
-        return this.getKey().split('_')[0];
+        return this.getKey();
     }
 
     getTransferId() {
-        return this.getKey().split('_')[1];
+        return (this.getContent() as IProcessPartyInfoCallbackCmdEvtData['content']).transferId;
```

Comment:
We must think of a better way to handle this. We are relying on "hard-coded" map values here. 

We spend a lot of effort in defining enums, but here we do not....but we use types, yet sub-type's cant be referenced otherwise except in a map....thus should we consider adding an enum to the type?

Reply:
Hi @mdebarros , I don't see any problem referencing subtypes using map. We can also overcome this if we define a type for that subtype and use that here like `IProcessPartyInfoCallbackCmdEvtDataContent`. I feel there will be a lot of type names and confusing.